### PR TITLE
bridge: add custom permission function in common bridge

### PR DESCRIPTION
**Pull request by khairul-alam-licon**
# Description
Add `requestCustomPermissions` function in `common-bridge.ts` to ask for the custom permissions from miniapps to Android, iOS or Native.

# Jira
[MINI-959](https://jira.rakuten-it.com/jira/browse/MINI-959)

# Checklist
- [ ] I wrote/updated tests for new/changed code.
- [x] I ran `yarn sdk buildSdk` and `yarn bridge buildBridge` without issues.
- [x] I ran `yarn sample prettify` and `yarn sample build` without issues.

### CODE REVIEW

File: js-miniapp-bridge/src/common-bridge.ts [hunk 1]
```diff
@@ -99,6 +99,21 @@ export class MiniAppBridge {
       );
     });
   }
+
+  /**
+   * Associating requestCustomPermissions function to MiniAppBridge object
+   * @param {string[]} permissionTypes, Array of type of custom permissions that is requested
```
Comment (ts-leojoseph-r) [inline js-miniapp-bridge/src/common-bridge.ts:105]: Array.<string> would be better. Also please try to add an example that could be much more helpful.

File: js-miniapp-bridge/src/common-bridge.ts [hunk 2]
```diff
@@ -99,6 +99,21 @@ export class MiniAppBridge {
...
+   */
+  requestCustomPermissions(permissionTypes: string[]) {
```
Comment (ts-leojoseph-r) [inline js-miniapp-bridge/src/common-bridge.ts:107]: I guess we don't need to specify the datatype that is passed.

Comment (ts-leojoseph-r) [inline js-miniapp-bridge/src/common-bridge.ts:107]: Also, I guess we need to have one more parameter for a description (which will be used by the host app while showing the custom permissions request)
`"The developer must specify a reason for each permission request."`
Please check [MINI-952](https://jira.rakuten-it.com/jira/browse/MINI-952)

Comment (minh-rakuten) [reply js-miniapp-bridge/src/common-bridge.ts:107]: In TypeScript, we need to specify the data type.
TS is defined as type object-oriented compile language which prevents the javascript any type style.

Comment (ts-leojoseph-r) [reply js-miniapp-bridge/src/common-bridge.ts:107]: Ah, I almost forgot that we changed to Typescript. My bad, sorry. @khairul-alam-licon please ignore.

Comment (khairul-alam-licon) [reply js-miniapp-bridge/src/common-bridge.ts:107]: I think, we can work with the "reason" parameter in a separate PR? Is it possible?

File: js-miniapp-bridge/src/common-bridge.ts (hunk 1 above)
Comment (khairul-alam-licon) [reply js-miniapp-bridge/src/common-bridge.ts:105]: I have added an example, let me to know if it's okay, or we can do something better.

File: js-miniapp-bridge/src/common-bridge.ts (hunk 2 above)
Comment (corycaywood) [reply js-miniapp-bridge/src/common-bridge.ts:108]: The `description` will passed within the string array, so I don't think anything needs to be done here.

File: js-miniapp-bridge/src/common-bridge.ts (hunk 1 above)
Comment (corycaywood) [reply js-miniapp-bridge/src/common-bridge.ts:105]: Yeah, I think this needs an example of the string array that should be passed.

**Review by corycaywood — APPROVED** [review]
Looks good 👍  But I think we need an example of how the params are passed to `requestCustomPermissoions`.

File: js-miniapp-bridge/src/common-bridge.ts (hunk 2 above)
Comment (ts-leojoseph-r) [reply js-miniapp-bridge/src/common-bridge.ts:108]: That's right @corycaywood , we actually changed the request  string object after the comment here. So I don't think any change is required here.

File: js-miniapp-bridge/src/common-bridge.ts [hunk 3]
```diff
@@ -99,6 +99,35 @@ export class MiniAppBridge {
...
+   *  {"name":"rakuten.miniapp.user.PROFILE_PHOTO", "description": "Reason to request for the custom permission"},
+   *  {"name":"rakuten.miniapp.user.CONTACT_LIST", "description": "Reason to request for the custom permission"}
+   * ]
+   *
+   * For eg., Miniapps can pass the array of invalid custom permissions as following
+   * [
+   *  {"name":"any unknown name", "description": "Reason to request for the custom permission"}
+   * ]
+   */
+  requestCustomPermissions(permissionTypes: [string, string]) {
```
Comment (corycaywood) [inline js-miniapp-bridge/src/common-bridge.ts:121]: I think that `[string, string]` defines an Array that contains two string values, i.e. `['first string', 'second string']`. But this is supposed to be an array that contains objects, right? So I think it should be something like this:
```suggestion
  requestCustomPermissions(permissionTypes: Array<{name: string, description: string}>) {
```

File: js-miniapp-bridge/src/common-bridge.ts [hunk 4]
```diff
@@ -99,6 +99,35 @@ export class MiniAppBridge {
...
+   *  {"name":"rakuten.miniapp.user.USER_NAME", "description": "Reason to request for the custom permission"},
+   *  {"name":"rakuten.miniapp.user.PROFILE_PHOTO", "description": "Reason to request for the custom permission"},
+   *  {"name":"rakuten.miniapp.user.CONTACT_LIST", "description": "Reason to request for the custom permission"}
+   * ]
+   *
+   * For eg., Miniapps can pass the array of invalid custom permissions as following
+   * [
+   *  {"name":"any unknown name", "description": "Reason to request for the custom permission"}
+   * ]
+   */
```
Comment (minh-rakuten) [inline js-miniapp-bridge/src/common-bridge.ts:120]: I think we can remove those lines.

File: js-miniapp-bridge/src/common-bridge.ts (hunk 3 above)
Comment (minh-rakuten) [reply js-miniapp-bridge/src/common-bridge.ts:121]: How about creating an object for this? `Array<CustomPermission>`?

Comment (corycaywood) [reply js-miniapp-bridge/src/common-bridge.ts:121]: Yeah, making a `CustomPermission` type is even better.

**Review by corycaywood — APPROVED** [review]
👍

File: js-miniapp-bridge/src/common-bridge.ts (hunk 3 above)
Comment (khairul-alam-licon) [reply js-miniapp-bridge/src/common-bridge.ts]: Thanks, I have updated.

**Review by minh-rakuten — APPROVED** [review]
👍 👍

**Pull request merged**